src/Quereum.sol

```diff
@@ -1,14 +1,96 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
-contract Counter {
-    uint256 public number;
+contract Quereum {
+    struct Question {
+        string question; // the actual question text
+        uint256 expirationTime; // let's use UTC time?
+        uint256 status; // 0 for active 1 for expired 2 for closed
+        address user; // address of the user who posted the question
+        uint256 reward; // reward for answering the question
+        uint256[] responses; // array of response ids
+        bool rewardAllocated; // if the reward has been allocated or not
+    }
+
+    // Mapping of user addresses to their names/balances/questions
+    mapping(address => string) private accounts;
+    mapping(address => uint256) private balances;
+    Question[] private questions;
+    mapping(address => uint256[]) private userQuestions;
+
+    // Register a new user with a chosen name
+    function register(
+        address userAddress,
+        string memory name
+    ) public returns (bool) {
+        require(
+            bytes(accounts[userAddress]).length == 0,
+            "User already registered"
+        );
+        accounts[userAddress] = name;
+        balances[userAddress] = 0; // initialize balance to 0
+        return true;
+    }
+
+    // add balance to the user's account
+    function addBalance(uint256 amount) public returns (bool) {
+        require(bytes(accounts[msg.sender]).length > 0, "User not registered");
+        balances[msg.sender] += amount;
+        return true;
+    }
+
+    // helper function to calculate the total locked reward for a user
+    function getTotalLockedReward(
+        address userAddress
+    ) internal view returns (uint256) {
+        uint256 totalLockedReward = 0;
+        // get the ids of all questions posted by the user
+        uint256[] memory userQuestionIds = userQuestions[userAddress];
+
+        for (uint256 i = 0; i < userQuestionIds.length; i++) {
+            Question storage question = questions[userQuestionIds[i]]; // get the question
+            if (question.status == 0 && !question.rewardAllocated) {
+                totalLockedReward += question.reward; // add the reward to the total locked reward
+            }
+        }
+
+        return totalLockedReward;
+    }
+
+    // post a new question with a reward
+    function postQuestion(
+        string memory questionText,
+        uint256 expirationTime,
+        uint256 reward
+    ) public returns (bool) {
+        require(bytes(accounts[msg.sender]).length > 0, "User not registered");
+
+        // calculate all the rewards in the questionst the user posted to ensure they have a sufficient balance
+        uint256 totalLockedReward = getTotalLockedReward(msg.sender);
+        require(
+            balances[msg.sender] >= totalLockedReward + reward,
+            "Insufficient balance for reward"
+        );
+

```

Comment:
We need to add a require statement to make sure that expirationTime is after the current time.